### PR TITLE
Convert Overview.Rnw to Overview.Rmd

### PR DESCRIPTION
**Pull request by Rohit-Satyam**
Asking for review from @Bioconductor/sweave2rmd @jwokaty @mcarlsn for vignette Overview.Rnw conversion. Below attached is the HTML output of the vignette.
[Overview.zip](https://github.com/Bioconductor/ShortRead/files/11369955/Overview.zip)

### CODE REVIEW

File: vignettes/Overview.Rmd
Comment (mtmorgan) [inline vignettes/Overview.Rmd]: Insert a blank line here so that the HTML contains a new paragraph, like the Rnw.

There are other places in the Rmd that need to be modified.

Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done

Comment (mtmorgan) [reply vignettes/Overview.Rmd]: As mentioned, there are other locations where a similar change needs to be made, so that the paragraphs of the Rmd vignette are the same as the paragraphs of the Rnw vignette.

File: DESCRIPTION
Comment (jwokaty) [inline DESCRIPTION]: Lines 7 and 9 should be indented to make it easier to read.

```
Authors@R: c(
        person("Rohit", "Satyam", role = "ctb",
                comment = "Converted Overview.Rnw vignette from Sweave to RMarkdown / HTML."),
        person("Bioconductor Package Maintainer",
                email = "maintainer@bioconductor.org", role = "cre"),
```

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: Use the following for date:

```
date: 'Modified: October 21, 2013. Compiled `r format(Sys.Date(), "%B %d, %Y")`'
```

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: I believe the default for `eval` is already `TRUE`, so it isn't needed. For consistency, I recommend always using `TRUE`/`FALSE` rather than `T`/`F`. I am not an R expert, but https://www.r-bloggers.com/2012/12/r-tip-avoid-using-t-and-f-as-synonyms-for-true-and-false/ says that `T` and `F` are symbols that can be redefined. 

I would also add a space after the comma and note we typically favor blank spaces before and after code fences to improve readability.

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: Insert a blank line before line 294 that starts `Suppose the task is to read column 9 ...`

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: Insert a blank before line 302 that starts "Important arguments for ..."

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: Insert a blank line before line 336 that begins `There are several advantages to reading ...`

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: ShortRead, Biostrings, and BSgenome should be links with, for example,

```
`r Biocpkg("ShortRead")`
```

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: Insert a blank line before line 384 that begins `The distribution component ...`

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: Insert a blank line before line 425 that begins `Users wanting to use **R** ...`

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: The table should be after the paragraph that begins `The default size for both streams ...` at line 104.

I know it took some time to translate the latex into an R Markdown table. Thanks for that. You can also make the links and make it easier to read as follows.

```
Functionality is summarized in Table \@ref(tab:table).

|                                  |                      |                                            |
| -------------------------------- | -------------------- | ------------------------------------------ |
| **Input**                        |                      |                                            |
|                                  | `FastqStreamer`      | Iterate through FASTQ files in chunks      |
|                                  | `FastqSampler`       | Draw random samples from FASTQ files       |
|                                  | `readFastq`          | Read an entire FASTQ file into memory      |
|                                  | `writeFastq`         | Write FASTQ objects to a connection (file) |
|                                  | `countFastq`         | Quickly count FASTQ records in files       |
| **Sequence and quality summary** |                      |                                            |
|                                  | `alphabetFrequency`  | Nucleotide or quality score use per read   |
|                                  | `alphabetByCycle`    | Nucleotide or quality score use by cycle   |
|                                  | `alphabetScore`      | Whole-read quality summary                 |
|                                  | `encoding`           | Character / ‘phred’ score mapping          |
| **Quality assessment**           |                      |                                            |
|                                  | `qa`                 | Visit FASTQ files to collect QA statistics |
|                                  | `report`             | Generate a quality assessment report       |
| **Filtering and trimming**       |                      |                                            |
|                                  | `srFilter`           | Pre-defined and bespoke filters            |
|                                  | `trimTails`, etc.    | Trim low-quality nucleotides               |
|                                  | `narrow`             | Remove leading / trailing nucleotides      |
|                                  | `tables`             | Summarize read occurrence                  |
|                                  | `srduplicated`, etc. | Identify duplicate reads                   |
|                                  | `filterFastq`        | Filter reads from one file to another      |

: (\#tab:table) Key functions for working with FASTQ files
```

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: Use

   `r Biocpkg("ShortRead")`

**Review by jwokaty — CHANGES_REQUESTED** [review]
@Rohit-Satyam Thanks for another PR. This is the checklist we're using to evaluate your PR. I've made some inline comments as well that need resolution before we can merge.

- [x] the `.Rmd` file knits to `HTML`
- [x] `R CMD build` runs without errors or timeouts
- [x] the tarball from `R CMD build` contains the `HTML` (check with the
  following by substituting the package name and vignette name `tar ztf
  package_name.tar.gz | grep 'doc/vignette_name'`)
- [x] the `.Rnw` file has been removed
- In the `DESCRIPTION` file
  - [x] `BiocStyle` and `knitr` are listed in `Suggests`
  - [x] the line `VignetteBuilder: knitr` exists
  - [ ] any added lines use the same spacing and indents as the existing document
- If agreed with the Maintainer
  - [x] the contributor is in the author list in the `DESCRIPTION` file*.
  - [x] the contributor is in the author list in the vignette's YAML*.
- If this pull request involves converting from separate `Author` and  `Maintainer`
  lines to the `Authors@R` vector, please ensure that
  - [x] the `Authors@R` vector includes the maintainer as specified with `role='cre'`.
  - [x] the `Maintainer` line is completely removed.
- [x] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [ ] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request
- [x] the R Markdown file is representative of the Sweave document and follows
  best practices, such as replacing links to Bioconductor packages with calls
  to BiocStyle's `Biocpkg()`
- [x] Only files necessary for the conversion are included in the pull request.
- [ ] Long lines have been broken up and reformatted, you can achieve this by
  selecting the text in your script and then clicking on `Code > Reflow Comment`
  in R Studio.
- [x] Code blocks are surrounded by a blank line, and there are no blank lines
  before the start or after the end of the code within the code block. 

* Refer to the contribution guide for acceptable formats

File: DESCRIPTION
Comment (Rohit-Satyam) [reply DESCRIPTION]: Done

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Removed `eval` and added space before the code block. Sorry to miss this.

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Sure. Improved.

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Thanks. I wasn't aware how to link markdown tables in Rmd. Glad to have learned new thing.

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done

File: vignettes/Overview.Rmd
Comment (jwokaty) [reply vignettes/Overview.Rmd]: The original has a Modified and a Compiled date.

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: I looked at this section again and I think the .Rnw splits the short paragraph. It's also missing the link to table 2, which should be like table 1. Could you change this section to

```
The result of data input is an instance of class `ShortReadQ` 
(Table \@ref(tab:table2)). This class contains reads, their quality scores, and
optionally the id of the read.

|                    |                                                   |
| ------------------ | ------------------------------------------------- |
| DNAStringSet       | (`r Biocpkg("Biostrings")`) Short read sequences  |
| FastqQuality, etc. | Quality encodings                                 |
| ShortReadQ         | Reads, quality scores, and ids                    |

: (\#tab:table2) Primary data types in the `r Biocpkg("ShortRead")` package.

```

File: vignettes/Overview.Rmd
Comment (jwokaty) [reply vignettes/Overview.Rmd]: I think we've caught the other locations in other comments.

**Review by jwokaty — CHANGES_REQUESTED** [review]
@Rohit-Satyam We just need 2 changes and I think this will be ready to merge.

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done.

Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: The paragraphs were split to match the formatting of the PDF document. If you find that there are more paragraphs that needs to be fixed please let me know.

File: vignettes/Overview.Rmd
Comment (jwokaty) [inline vignettes/Overview.Rmd]: Add a space after the period.

File: vignettes/Overview.Rmd
Comment (jwokaty) [reply vignettes/Overview.Rmd]: This still needs to be fixed.

**Review by jwokaty — CHANGES_REQUESTED** [review]
The date still needs to be corrected with a compiled and modified date. We also need a space after a sentence.

I just want to caution against editing the text. Since this package is maintained by the core, I can make decisions about accepting edits to the text; however,  another maintainer may disagree, so I usually advise contributors to discuss it with the maintainer. Then the maintainer can choose, you haven't done any extra work, and your PR will still be on track. 

Even if the maintainer was okay with the changes, especially if there were a lot, I might still recommend making text edits in a completely separate PR.

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done. Changed the date format using `%B %d, %Y` notation.

File: vignettes/Overview.Rmd
Comment (Rohit-Satyam) [reply vignettes/Overview.Rmd]: Done

**Comment by Rohit-Satyam** [comment]
Hi @jwokaty. I didn't perform major grammatical changes, just the one suggested by grammar checker in R and few punctuations here and there and some misspelled words. I hope that shouldn't be a problem.

File: vignettes/Overview.Rmd
Comment (sonali8434) [reply vignettes/Overview.Rmd]: please change this
```
date: "`r format(Sys.Date(), '%B %d, %Y')`"
```
 to 

```
date: 'Modified: October 21, 2013. Compiled `r format(Sys.Date(), "%B %d, %Y")`'
```

**Pull request merged**
